compiler/luci-interpreter/src/kernels/ArgMax.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/ArgMax.h"
+#include "kernels/Utils.h"
+#include <tensorflow/lite/kernels/internal/optimized/optimized_ops.h>
+
+namespace luci_interpreter
+{
+namespace kernels
+{
+
+ArgMax::ArgMax(const Tensor *input, const Tensor *axis, Tensor *output, const ArgMaxParams &params)
+    : KernelWithParams<ArgMaxParams>(params), _input(input), _axis(axis), _output(output)
+{
+}
+
+void ArgMax::configure()
+{
+  assert(_axis->shape().num_elements() == 1);
+  assert(_axis->element_type() == DataType::S32 || _axis->element_type() == DataType::S64);
+  assert(_input->shape().num_dims() >= 1);
+  const Shape &input_shape = _input->shape();
+  const int num_dims = input_shape.num_dims();
+  Shape output_shape(num_dims - 1);
+  const auto *axis = getTensorData<int32_t>(_axis);
```

Comment:
Axis value can be negative. It should be normalized like this:
```C++
int axis_value = *getTensorData<int32_t>(_axis);
if (axis_value < 0)
  _axis_value += num_dims;
```
Please move the `assert` that checks the number of elements in `_axis` closer to this code. It is unclear that there should be only one element in the tensor.

Reply:
@s-barannikov I fix it for the case when axis is negative, but last sentence make me confused. 
> It is unclear that there should be only one element in the tensor.
 I found codes in tensorflow lite kernel
```C++
// Make sure the axis is only 1 dimension.
TF_LITE_ENSURE_EQ(context, NumElements(axis), 1);
```
So, I add like below
```C++
assert(_axis->shape().num_elements() == 1);
```
If any checking logic for this case, please guide me.

Reply:
What I meant is that it would be more clear if this assert is located close to extracting the only element:
```C++
assert(_axis->shape().num_elements() == 1);
int axis = getTensorData<int32_t>(_axis)[0];
if (axis < 0)
  axis += num_dims;
assert(axis >= 0 && axis < num_dims);
```

Reply:
oh, I got what you mean. I will update it.